functions/snippets/src/main/java/com/example/functions/SlackSlashCommand.java

```diff
@@ -0,0 +1,172 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.functions;
+
+import com.github.seratch.jslack.app_backend.SlackSignature;
+import com.google.api.client.googleapis.javanet.GoogleNetHttpTransport;
+import com.google.api.client.json.jackson2.JacksonFactory;
+import com.google.api.client.util.ArrayMap;
+import com.google.api.services.kgsearch.v1.Kgsearch;
+import com.google.cloud.functions.HttpFunction;
+import com.google.cloud.functions.HttpRequest;
+import com.google.cloud.functions.HttpResponse;
+import com.google.gson.Gson;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonObject;
+import java.io.BufferedWriter;
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.security.GeneralSecurityException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+
+public class SlackSlashCommand implements HttpFunction {
+
+  private Kgsearch kgClient;
+  private static String API_KEY;
+  private static String SLACK_SECRET;
+  private static final Logger LOGGER = Logger.getLogger(HelloHttp.class.getName());
+  private SlackSignature.Verifier verifier;
+  private Gson gson = new Gson();
+
+  public SlackSlashCommand() throws IOException, GeneralSecurityException {
+    kgClient = new Kgsearch.Builder(
+        GoogleNetHttpTransport.newTrustedTransport(), new JacksonFactory(), null).build();
+
+    // Read + parse config file
+    Path configPath = Path.of(System.getProperty("user.dir"), "config.json");
```

Comment:
OK, but isn't this against 12 factor?  (Shouldn't config come from EnvVar's)?

Reply:
I'm not sure the Java runtime (easily) supports file uploads anyway - so that or properties in `pom.xml` are probably the way to go here.

(The other samples use `config.json` files, and I was copying what they did.)